tests/Helper/HeaderHelperTest.php

```diff
@@ -134,37 +150,75 @@ public function testSortedAcceptTypesFromRequest(RequestInterface $request, arra
+                'a="tes\'t";test="\'test\'";test2="\\"quoted\\" test"',null, null,
+                ['a' => 'tes\'t', 'test' => '\'test\'', 'test2' => '"quoted" test']
+            ],
+            'slashes' => [
+                'a="\\t\\e\\s\\t";b="te\\\\st";c="\\"\\"',null, null,
+                ['a' => 'test', 'b' => 'te\\st', 'c' => '"\\']
+            ],
+            'specChars' => ['*=test;test=*',null, null, ['*' => 'test', 'test' => '*']],
+            'specChars2' => ['param*1=a;param*2=b',null, null, ['param*1' => 'a', 'param*2' => 'b']],
+            'numbers' => ['a=8888;b="999"',null, null, ['a' => '8888', 'b' => '999']],
+            'invalidQuotes2' => ['a="test',null, null, null, \InvalidArgumentException::class],
+            'invalidQuotes3' => ['a=test"',null, null, null, \InvalidArgumentException::class],
+            'invalidQuotes4' => ['a=te"st',null, null, null, \InvalidArgumentException::class],
+            'invalidEmptyValue' => ['a=b; c=',null, null, null, \InvalidArgumentException::class],
+            'invalidEmptyParam' => ['a=b; ;c=d',null, null, null, \InvalidArgumentException::class],
+            'semicolonAtEnd' => ['a=b;',null, null, null, \InvalidArgumentException::class],
+            'comma' => ['a=test,test',null, null, null, \InvalidArgumentException::class],
+
+            # true syntax
+            // 'spaces2' => [' a = b ; c = "d" ',null, null, ['a' => 'b', 'c' => 'd']],
+            'case' => ['A=TEST;TEST=B',true, false, ['a' => 'TEST', 'test' => 'B']],
+            'case2' => ['A=TEST;TEST=B',false, false, ['A' => 'TEST', 'TEST' => 'B']],
+            'case3' => ['A=TEST;TEST=B',false, true, ['A' => 'test', 'TEST' => 'b']],
+            // 'spaces1' => ['a=b; c="d" ',null, null, ['a' => 'b', 'c' => 'd']],
+            'spaces3' => ['a=b c',null, null, null, \InvalidArgumentException::class],
+            'percent' => ['a=%1;b="foo-%32-bar"',null, null, ['a' => '%1', 'b' => 'foo-%32-bar']],
+
+            # Invalid syntax but most browsers take a first parameter
+            // 'sameName' => ['a=T1;a="T2"',null, null, ['a' => 'T1']],
+            // 'sameNameCase' => ['aa=T1;Aa="T2"',null, null, ['aa' => 'T1']],
+
+            # Invalid syntax but most browsers accept the unquoted value with warn
+            # What is better for us to do?
+            'brokenToken' => ['a=foo[1](2).html',null, null, null, \InvalidArgumentException::class],
+
+            'brokenSyntax1' => ['a==b',null, null, null, \InvalidArgumentException::class],
+            'brokenSyntax2' => ['a *=b',null, null, null, \InvalidArgumentException::class],
+            # Invalid syntax but most browsers accept the umlaut with warn
+            'brokenToken2' => ['a=foo-ä.html',null, null, ['a' => 'foo-ä.html']],
+            # Invalid syntax but most browsers accept the umlaut with warn
+            'brokenToken3' => ['a=foo-Ã¤.html',null, null, ['a' => 'foo-Ã¤.html']],
+
+            # RFC 2231/5987 Encoding: Character Sets
```

Comment:
I think this usually only appears in response. #174

With low priority, I can imagine a separate issue.

Reply:
I also think that it is not practical to implement it now

Reply:
If we aren't going to implement it it is better to remove commented out tests as well.

Reply:
removed and saved https://github.com/yiisoft/yii-web/issues/174#issuecomment-568657384